extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts

```diff
@@ -0,0 +1,367 @@
+	let credObj = { credentials: {} };
+	let pairs = creds.split(',');
+	// handle escaped commas in a browser-agnostic way by pushing to the next
+	let validPairs: string[] = [];
+	for (let i = 0; i < pairs.length; i++) {
+		if (i < (pairs.length - 1) && pairs[i].match(/(?!\\).*\\$/)) {
+			// Since this was meant to be an escaped comma, restore it
+			pairs[i + 1] = `${pairs[i]},${pairs[i + 1]}`;
+		} else {
+			validPairs.push(pairs[i]);
+		}
+	}
+
+	validPairs.forEach(pair => {
+		const formattingErr = localize('mount.err.formatting', "Bad formatting of credentials at {0}", pair);
+		try {
+			// # remove escaped characters for ,
+			pair = pair.replace('\\,', ',').trim();
+			let firstEquals = pair.indexOf('=');
+			if (firstEquals <= 0 || firstEquals >= pair.length) {
+				throw new Error(formattingErr);
+			}
+			let key = pair.substring(0, firstEquals);
+			let value = pair.substring(firstEquals + 1);
+			credObj.credentials[key] = value;
+		} catch (err) {
+			throw new Error(formattingErr);
+		}
+	});
+	return credObj;
+}
+
+export interface DialogProperties {
+	url?: string;
+	auth?: AuthType;
+	username?: string;
+	password?: string;
+}
+
+export interface MountHdfsProperties extends DialogProperties {
+	hdfsPath?: string;
+	remoteUri?: string;
+	credentials?: string;
+}
+
+abstract class HdfsDialogModelBase<T extends DialogProperties> {
+	protected _canceled = false;
+	private _authTypes: azdata.CategoryValue[];
+	constructor(
+		public props: T
+	) {
+		if (!props.auth) {
+			this.props.auth = 'basic';
+		}
+	}
+
+	public get authCategories(): azdata.CategoryValue[] {
+		if (!this._authTypes) {
+			this._authTypes = [getAuthCategory('basic'), getAuthCategory('integrated')];
+		}
+		return this._authTypes;
+	}
+
+	public get authCategory(): azdata.CategoryValue {
+		return getAuthCategory(this.props.auth);
+	}
+
+	public async onComplete(props: T): Promise<void> {
+		try {
+			this.props = props;
+			await this.handleCompleted();
+
+		} catch (error) {
+			// Ignore the error if we cancelled the request since we can't stop the actual request from completing
+			if (!this._canceled) {
+				throw error;
+			}
+		}
+	}
+
+	protected abstract handleCompleted(): Promise<void>;
+
+	public async onError(error: ControllerError): Promise<void> {
+		// implement
+	}
+
+	public async onCancel(): Promise<void> {
+		this._canceled = true;
+	}
+
+	protected createController(): ClusterController {
+		return new ClusterController(this.props.url, this.props.auth, this.props.username, this.props.password, true);
+	}
+
+}
+
+export class MountHdfsDialogModel extends HdfsDialogModelBase<MountHdfsProperties> {
+	private credentials: {};
+
+	constructor(props: MountHdfsProperties) {
+		super(props);
+	}
+
+	protected async handleCompleted(): Promise<void> {
+		if (this.props.auth === 'basic') {
+			// Verify username and password as we can't make them required in the UI
+			if (!this.props.username) {
```

Comment:
Why not?

Reply:
Covered in previous PR - in a FormContainer I don't know how to dynamically update the required status?